models/DataObject/ClassDefinition/Data/Extension/RelationFilterConditionParser.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @category    pimcore
+ * @date        25/06/2021
+ * @author      Filip Szenborn <fszenborn@divante.co>
+ */
+declare(strict_types=1);
+
+namespace Pimcore\Model\DataObject\ClassDefinition\Data\Extension;
+
+/**
+ * Trait RelationFilterConditionParser
+ * @package Pimcore\Model\DataObject\ClassDefinition\Data\Extension
+ */
+trait RelationFilterConditionParser
+{
+    /**
+     * Parses filter value of a relation field and creates the filter condition
+     * @param string $value
+     * @param string $operator
+     * @param string $name
+     * @return string
+     */
+    public function getRelationFilterCondition($value, $operator, $name)
+    {
+        if ($operator == '=') {
+            return '`' . $name . '` LIKE ' . "'%," . $value . ",%'";
```

Comment:
we should consider type as well for relation filters

Imagine a case where a relation field has both document & object relations having same id then result set will wrong in case of filter based on just one type.

Reply:
please check this files for the piece that considers type as well: 
bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyRelation.js
bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyRelation.js

Reply:
for manyToMany and advancedManyToMany it is considered
but for ManyToOne it isn't, you're right.